File: manager-web/src/main/resources/static/js/service/goodsService.js
//服务层
app.service('goodsService',function($http){
	    	

	    //查找所有
        this.findAll = function () {
            return $http.get("./goods");
        }
    
        //分页查找
        this.findPage = function (pageNo, pageSize) {
            return $http.get("./goods/" + pageNo + "/" + pageSize);
        }
    
        //添加品牌数据
        this.save = function (entity) {
            return $http.post("./goods", entity);
        }
    
        //修改品牌数据
        this.update = function (entity) {
            return $http.put("./goods", entity);
        }
    
        //删除品牌数据
        this.delete = function (selectIds) {
            return $http.delete("./goods?ids=" + selectIds);
        }
    
        //模糊查询品牌数据
        this.search=function (pageNo, pageSize,entity) {
            return $http.post("./goods/search/" + pageNo + "/" + pageSize, entity);
        }
  	
});

File: sellergoods-interface/src/main/java/com/gm/service/GoodsService.java
package com.gm.service;

import com.gm.common.PageResult;
import com.gm.common.Result;
import com.gm.pojo.Goods;
import com.gm.pojo.TbGoods;

public interface GoodsService {
    Result findAll();

    PageResult findPage(Integer pageNo, Integer pageSize);

    Result findPage(TbGoods goods, Integer pageNo, Integer pageSize);

    Result saveGoods(Goods goods);

    Result updateGoods(Goods goods);

    Result deleteGoods(Long[] ids);

    Result deleteOneGoods(Long id);
}
